### PR TITLE
Add support for MJCF <exclude> tag for collision filtering

### PR DESCRIPTION
**Pull request by adenzler-nvidia**
Implement parsing of <exclude> elements from the <contact> section in MJCF files. The <exclude> tag allows specifying pairs of bodies whose shapes should not collide, matching MuJoCo's built-in collision filtering mechanism.

Features:
- Parse body1/body2 attributes from <exclude> elements
- Find all shapes belonging to the specified bodies
- Add all shape pairs to collision filter pairs
- Gracefully handle missing bodies or attributes with warnings (verbose mode)
- Support verbose output for debugging
- Error handling consistent with existing <pair> parsing (warnings, not exceptions)

Tests:
- Verify exclude tag correctly filters collisions between specified bodies
- Test verbose output includes exclude parsing information
- Test graceful handling of missing body references
- Test graceful handling of missing attributes
- Test that warnings are printed when verbose=True
- All tests verify model can be finalized successfully after parse errors

Also add guideline to AGENTS.md to exclude AI attribution from commit messages.

## Before your PR is "Ready for review"

- [x] Necessary tests have been added and new examples are tested (see `newton/tests/test_examples.py`)
- [x] Documentation is up-to-date
- [x] Code passes formatting and linting checks with `pre-commit run -a`

## Summary by CodeRabbit

* **New Features**
  * Collision exclusion: specify body pairs to ignore during collision checks, including support for hyphenated body names.

* **Documentation**
  * Updated contributor guidelines: require running pre-commit hooks and exclude AI attribution/co-authorship lines from commit messages.

* **Tests**
  * Added extensive tests and test assets covering exclusion parsing, hyphenated names, missing-body handling, verbose warnings, and various edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: newton/_src/utils/import_mjcf.py [hunk 1]
```diff
@@ -1521,6 +1521,47 @@ def get_site_body_and_anchor(site_name: str) -> tuple[int, wp.vec3] | None:
...
+            # Add all shape pairs from these bodies to collision filter
+            for shape1_idx in body1_shapes:
+                for shape2_idx in body2_shapes:
+                    builder.shape_collision_filter_pairs.append((shape1_idx, shape2_idx))
+
+            if verbose:
+                print(
+                    f"Parsed collision exclude: {body1_name} ({len(body1_shapes)} shapes) <-> "
+                    f"{body2_name} ({len(body2_shapes)} shapes), added {len(body1_shapes) * len(body2_shapes)} filter pairs"
+                )
```
[the remark this answers is not in the document]
Comment (adenzler-nvidia) [reply newton/_src/utils/import_mjcf.py:1567]: @coderabbitai can you apply this yourself?
